tests/integration/backward_compatible/serialization/compact_test.py

```diff
@@ -0,0 +1,854 @@
+           http://www.hazelcast.com/schema/config/hazelcast-config-5.0.xsd">
+            <serialization>
+                    <compact-serialization enabled="true" />
+            </serialization>
+        </hazelcast>
+        """
+
+        cls.cluster = cls.create_cluster(cls.rc, config)
+        cls.member = cls.cluster.start_member()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.rc.terminateCluster(cls.cluster.id)
+        cls.rc.exit()
+
+    def tearDown(self) -> None:
+        self.shutdown_all_clients()
+
+
+class CompactTest(CompactTestBase):
+    def test_write_then_read_with_all_fields(self):
+        serializer = SomeFieldsSerializer.from_kinds(FIELD_KINDS)
+        self._write_then_read(FIELD_KINDS, REFERENCE_OBJECTS, serializer)
+
+    def test_write_then_read_with_no_fields(self):
+        serializer = SomeFieldsSerializer.from_kinds([])
+        self._write_then_read([], {}, serializer)
+
+    def test_write_then_read_with_just_var_sized_fields(self):
+        serializer = SomeFieldsSerializer.from_kinds(VAR_SIZED_FIELD_KINDS)
+        self._write_then_read(VAR_SIZED_FIELD_KINDS, REFERENCE_OBJECTS, serializer)
+
+    def test_write_then_read_with_just_fix_sized_fields(self):
+        serializer = SomeFieldsSerializer.from_kinds(FIX_SIZED_FIELD_KINDS)
+        self._write_then_read(FIX_SIZED_FIELD_KINDS, REFERENCE_OBJECTS, serializer)
+
+    @parameterized.expand(
+        [
+            ("small", 1),
+            ("medium", 20),
+            ("large", 42),
+        ]
+    )
+    def test_write_then_read_object(self, _, array_item_count):
+        reference_objects = {
+            FieldKind.ARRAY_OF_STRING: ["x" * (i * 100) for i in range(1, array_item_count)],
+            FieldKind.INT32: 32,
+            FieldKind.STRING: "hey",
+        }
+        reference_objects[FieldKind.ARRAY_OF_STRING].append(None)
+        serializer = SomeFieldsSerializer.from_kinds(list(reference_objects.keys()))
+        self._write_then_read(list(reference_objects.keys()), reference_objects, serializer)
+
+    @parameterized.expand(
+        [
+            ("0", 0),
+            ("1", 1),
+            ("8", 8),
+            ("10", 10),
+            ("100", 100),
+            ("1000", 1000),
+        ]
+    )
+    def test_write_then_read_boolean_array(self, _, item_count):
+        reference_objects = {
+            FieldKind.ARRAY_OF_BOOLEAN: [
+                random.randrange(0, 10) % 2 == 0 for _ in range(item_count)
+            ]
+        }
+        serializer = SomeFieldsSerializer.from_kinds(list(reference_objects.keys()))
+        self._write_then_read(list(reference_objects.keys()), reference_objects, serializer)
+
+    @parameterized.expand(
+        [
+            ("0", 0),
+            ("1", 1),
+            ("8", 8),
+            ("10", 10),
+            ("100", 100),
+            ("1000", 1000),
+        ]
+    )
+    def test_write_and_read_with_multiple_boolean_fields(self, _, field_count):
```

Comment:
it is actually used in test names by the pytest, which makes the inspection easier. I use `_` in the parameter list because the name is not needed in the test